src/app/server/Server.cpp

```diff
@@ -66,6 +66,21 @@ constexpr bool isRendezvousBypassed()
 #endif
 }
 
+void StopEventLoop(intptr_t arg)
+{
+    LogErrorOnFailure(chip::DeviceLayer::PlatformMgr().StopEventLoopTask());
+}
+
+void DispatchShutDownEvent(intptr_t arg)
+{
+    // The ShutDown event SHOULD be emitted by a Node prior to any orderly shutdown sequence.
```

Comment:
There should be comments here indicating that delivery of the shut-down event to any subscribers is best-effort only. There are many situations in which the shut-down event likely won't be delivered to subscribers.